### ISSUE TITLE
Transitive extension requiredness

### ISSUE THREAD

**Issue by lexaknyazev** [KhronosGroup/glTF#1906]
Here's how the spec defines `extensionsRequired`:
> All glTF extensions required to load and/or render an asset must be listed in the top-level `extensionsRequired` array

We may want to elaborate a bit more there to cover potentially ambiguous cases like the following asset.
```json
{
    "asset": {
        "version": "2.0"
    },
    "extensionsUsed": [
        "KHR_texture_basisu", "KHR_materials_sheen"
    ],
    "extensionsRequired": [
        "KHR_texture_basisu"
    ],
    "materials": [
        {
            "extensions": {
                "KHR_materials_sheen": {
                    "sheenColorTexture": 0
                }
            }
        }
    ],
    "textures": [
        {
            "extensions": {
                "KHR_texture_basisu": {
                    "source": 0
                }
            }
        }
    ]
}
```

Here, the `KHR_texture_basisu` extension is required because there's no fallback image provided. However, the extended texture is only used by a material extension that could be completely ignored (including the texture) by non-supporting clients.

**Comment by emackey** [comment]
I see your point, but I think this is a case where simpler is better.  Most implementations can't handle a blank `texture` source, and we wouldn't want to count on a non-basisu implementation to process or ignore that texture correctly, with or without support for sheen.

So in my mind, this is correct as shown above.  Basisu is required (given there's no fallback source), and sheen is optional.  The idea that you could safely ignore the basisu part by dropping sheen support is too much big-picture thought for most simple-minded loaders.  The part of the loader that ingests textures typically can't see enough of the rest of the file to know that it can sidestep a problem like this.

**Comment by lexaknyazev** [comment]
I agree with the "simpler is better" approach but the spec as it's written currently doesn't suggest that since the BasisU extension is not technically "required to load and/or render an asset" given that supporting sheen is optional.

What do you think about revising the spec language to something along the lines of
> All glTF extensions that _do not define fallback behavior_ must be listed in the top-level `extensionsRequired` array

In other words, delegating the "required to load and/or render an asset" part to the extension specs and updating the latter where needed.

**Comment by emackey** [comment]
Ah, OK.  But I don't want this misinterpreted on the sheen side, either.  Sheen, like all PBR Next extensions, is implicitly optional, and the "fallback behavior" is simply to render without the sheen effect.  The glTF file doesn't need to supply any explicit fallback instructions given in the file itself.

Perhaps this is something that can/should be addressed directly in the spec for sheen and other extensions?  The sheen spec could say that it is always listed in `extensionsUsed` but never in `extensionsRequired`, and that the fallback behavior is to render without the sheen effect, as a typical core 2.0 implementation would do.  It looks like the basisu spec already has language describing when the extension should appear in `extensionsRequired`.

So perhaps we add the language @lexaknyazev suggested above, along with adding language about fallback behavior and used/required to each ratified extension?

**Comment by lexaknyazev** [comment]
Since the goal of `extensionsRequired` array is to provide an early exit for engines that might otherwise fail, let's address this explicitly in each extension.

As all PBR extensions are optional by design, their specs should mention that they shouldn't be required and therefore could be ignored by engines that don't support them. Another example is `KHR_mesh_quantization` that is always required (see the very end of its Overview section).
